src/index.d.ts

```diff
@@ -0,0 +1,229 @@
+import type { FunctionComponent, ReactElement } from 'react';
+import type { AutocompleteProps } from '@mui/material/Autocomplete/Autocomplete';
+import type {
+    ButtonProps,
+    RadioGroupProps,
+    SxProps,
+    TextFieldProps,
+} from '@mui/material';
+
+/**
+ * Section to export generated type declarations of .ts or .tsx files
+ */
+
+//TODO FM to remove (for the review)
+export { useTest } from './hooks/useTest';
+export { Test } from './utils/Test';
+
+export { useIntlRef } from './hooks/useIntlRef';
+
+/**
+ * Section to export manual type declarations of .js and .jsx files
+ */
+
+export const TopBar: FunctionComponent;
+
+export function logout(dispatch: any, userManagerInstance: any): Promise<any | undefined>;
+
+export const DARK_THEME: String, LIGHT_THEME: String;
+
+interface SnackInputs {
+    messageTxt?: string;
+    messageId?: string;
+    messageValues?: { [key: string]: string };
+    headerTxt?: string;
+    headerId?: string;
+    headerValues?: Record<string, string>;
+}
+
+interface UseSnackMessageReturn {
+    snackError: (snackInputs: SnackInputs) => void;
+    snackWarning: (snackInputs: SnackInputs) => void;
+    snackInfo: (snackInputs: SnackInputs) => void;
+}
+
+export function useSnackMessage(): UseSnackMessageReturn;
+
+type Input = string | number;
+type Options = Array<{
+    id: string;
+    label: string;
+}>;
+
+interface AutocompleteInputProps
+    extends Omit<
+        AutocompleteProps<
+            string | { id: string; label: string },
```

Comment:
Maybe we can extract a type for this `string | { id: string; label: string }` ?

Reply:
Definitely, will do in #357